test/e2e_env/multizone/meshservice/connectivity.go

```diff
@@ -0,0 +1,186 @@
+package meshservice
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/deployments/democlient"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+	"github.com/kumahq/kuma/test/framework/envs/multizone"
+)
+
+func Connectivity() {
+	namespace := "msconnectivity"
+	meshName := "msconnectivity"
+
+	BeforeAll(func() {
+		Expect(NewClusterSetup().
+			Install(MTLSMeshUniversal(meshName)).
+			Install(MeshTrafficPermissionAllowAllUniversal(meshName)).
+			Install(YamlUniversal(`
+type: HostnameGenerator
+name: kube-msconnectivity
+spec:
+  template: '{{ .DisplayName }}.{{ .Namespace }}.{{ .Zone }}.msconnectivity'
```

Comment:
could help to put `k8s` in the hostname so that it's super clear in the `It`s but feel free to ignore

```suggestion
  template: '{{ .DisplayName }}.{{ .Namespace }}.{{ .Zone }}.k8s.msconnectivity'
```